File: zd3.js
function isPrime(n) {
  if (n < 1) {
    return false;
  }
  for (let i = 2; i * i <= n; ++i) {
    if (n % i === 0) {
      return false;
    }
  }
  return true;
}

function main() {
  const numbers = [];
  let num;
  let sh;
  console.log("Введите количество цифр");
  sh = parseInt(prompt());
  
  console.log("Введите числа :");
  for (let i = 0; i < sh; i++) {
    num = parseInt(prompt());
    if (num > 0) {
      numbers.push(num);
    } else {
      console.log("Нужно вводить только числа:");
      return;
    }
  }
  
  console.log("Результаты проверки чисел на простоту:");
  for (let number of numbers) {
    if (isPrime(number)) {
      console.log(number + " - простое число");
    } else {
      console.log(number + " - не является простым числом");
    }
  }
}

main();

File: zd3.cpp

#include <iostream>
#include <vector>
using namespace std;

bool is_prime(int n)
{
    if (n < 1)
    {
        return false;
    }
    for (int i = 2; i * i <= n; ++i)
    {
        if (n % i == 0)
        {
            return false;
        }
    }
    return true;
}

int main() {
    setlocale(LC_ALL, "Russian");
    vector<int> numbers;
    int num;
    int sh;
    cout << "Введите количество цифр\n";
    cin >> sh;
    // Ввод чисел
    cout << "Введите числа :\n";
   
        for (int i = 0; i < sh; i++)
        {
            cin >> num;
            if (num > 0)
            {
                numbers.push_back(num);
            }
            else
            {
                cout << "Нужно вводить только числа:\n";

                return 0;
            }
        }
    

    // Проверка чисел на простоту
    cout << "Результаты проверки чисел на простоту:\n";
    for (int number : numbers)
    {
        if (is_prime(number))
        {
            cout << number << " - простое число\n";
        }
        else
        {
            cout << number << " - не является простым числом\n";
        }
    }


}
